### DISCUSSION TITLE
Restoring Data After Reconnect

### DISCUSSION THREAD

**Discussion by brentonhaerr** [socketio/socket.io#4610] [Q&A] [answered]
Sorry if this is a common issue, but I can't find an answer in the docs. 

After I reconnect a client due to a failure (not intentional), is ANY data maintained on the server-side? Right now it seems like I need to maintain a bunch of state on the client and resubmit it each time I reconnect, because the server treats it as a brand new connection. Is this correct, or should the automatic reconnection be able to recover its state and kind of 'pick up where it left off'?

**Comment by darrachequesne** [comment, marked as answer]
Hi!

> After I reconnect a client due to a failure (not intentional), is ANY data maintained on the server-side?

Currently, nothing is persisted on the server, but connection state recovery will be available in the next release: https://github.com/socketio/socket.io/commit/54d5ee05a684371191e207b8089f09fc24eb5107

Related: https://github.com/socketio/socket.io/discussions/4510

**Reply by brentonhaerr** [reply]
Great! I'm just working around it client-side for now. Interesting timing on the upgrade, though! Seems like a good feature. Is there any guidance on when the next release is happening (sorry if this is available somewhere, I'm new!).

**Reply by darrachequesne** [reply]
This should be released in a few days.
